src/Animals/Animal.java

```diff
@@ -0,0 +1,32 @@
+package Animals;
+
+public abstract class Animal {
+
+    private static int nextId;
+    private int id;
+    private String name;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public Animal() {}
```

Comment:
Seems this constructor is useless and harmful. It can create animals without ids